mlx/directives/item_matrix_directive.py

```diff
@@ -35,10 +35,15 @@ def perform_replacement(self, app, collection):
             collection (TraceableCollection): Collection for which to generate the nodes.
         """
         Rows = namedtuple('Rows', "sorted covered uncovered counters")
-        source_ids = collection.get_items(self['source'], attributes=self['filter-attributes'])
+        filters = {x: None for x in ['source', 'target']}
+        if self['filtertarget']:
+            filters['target'] = self['filter-attributes']
+        else:
+            filters['source'] = self['filter-attributes']
```

Comment:
My goal was to write it so that a future `filterintermediate` attribute would require as little code as possible, but if you still think your approach is better, I'm fine with it.